src/libultra/voice/voicecrc.c

```diff
@@ -1,3 +1,48 @@
+/**
+ * File: voicecrc.c
+ * Description: CRC check used by the Voice library functions in libultra.
+ *
+ * For general information about CRC, see the crc.c file (that's a lot of c's!).
+ */
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/voicecrc/__osVoiceContDataCrc.s")
+#define VOICE_CRC_LENGTH 8
+#define VOICE_CRC_GENERATOR 0x85
+
+/**
+ * This function is essentially the same as __osContDataCrc, but allows for a variable message length, specified by
+ * `numBytes`.
+ */
+u8 __osVoiceContDataCrc(u8* data, u32 numBytes) {
```

Comment:
```suggestion
u8 __osVoiceContDataCrc(u8* data, size_t numBytes) {
```

Is it possible to use size_t here? May not match.